### PR TITLE
feat: notifications service using gmail passwords.

### PR DESCRIPTION
**Pull request by Miguelito2774**
# Pull Request
 API for notifications

## Description

I implemented the email service using Mailgun's API to send notifications on the status of orders, either confirmation of success or instructions together.

## Type of Change

- [X] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality
      to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

**Comment by JorgeHB69** [comment]
Task linked: [CU-8689mr891 Order confirmation](https://app.clickup.com/t/8689mr891)

File: src/DistributionCenter.Services/NotificationService/Service/MailgunEmailService.cs [hunk 1]
```diff
@@ -0,0 +1,45 @@
+namespace DistributionCenter.Services.NotificationService.Service;
+
+using System.Text;
+using Concretes;
+using Interfaces;
+using RestSharp;
+
+public class MailgunEmailService(IRestClientWrapper client) : IEmailService
+{
+    private readonly string _domain = "sandbox843a364057074732963209c051944566.mailgun.org";
```
Comment (LuisHerbas98) [inline src/DistributionCenter.Services/NotificationService/Service/MailgunEmailService.cs]: This is sensitive information. It **MUST NOT** be in the code.

File: src/DistributionCenter.Services/NotificationService/Service/MailgunEmailService.cs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    private readonly string _apiKey = "cd178d76c14f2bd588ea6150ef763fcc-826eddfb-87d46213";
```
Comment (LuisHerbas98) [inline src/DistributionCenter.Services/NotificationService/Service/MailgunEmailService.cs]: Same Here

File: src/DistributionCenter.Services/NotificationService/Service/MailgunEmailService.cs [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    {
+        if (message == null)
+        {
+            throw new ArgumentNullException(nameof(message), "Message cannot be null");
+        }
+
+        RestRequest request = new("{domain}/messages", Method.Post);
+        _ = request.AddUrlSegment("domain", _domain);
+        _ = request.AddHeader("Authorization", "Basic " + Convert.ToBase64String(Encoding.ASCII.GetBytes($"api:{_apiKey}")));
+        _ = request.AddParameter("from", "mikyromesa100503@gmail.com");
```
Comment (LuisHerbas98) [inline src/DistributionCenter.Services/NotificationService/Service/MailgunEmailService.cs]: and here

File: src/DistributionCenter.Services/NotificationService/Service/MailgunEmailService.cs [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        RestResponse response = client.Execute(request);
+
+        if (!response.IsSuccessful)
+        {
+            string errorMessage = "An error occurred while sending the email.";
+            HtmlMessage errorNotification = HtmlMessage.CreateErrorNotification(errorMessage);
+            _ = client.Execute(new RestRequest("{domain}/messages", Method.Post)
+                .AddUrlSegment("domain", _domain)
+                .AddHeader("Authorization", "Basic " + Convert.ToBase64String(Encoding.ASCII.GetBytes($"api:{_apiKey}")))
+                .AddParameter("from", "mikyromesa100503@gmail.com")
```
Comment (LuisHerbas98) [inline src/DistributionCenter.Services/NotificationService/Service/MailgunEmailService.cs]: here

File: src/DistributionCenter.Services/NotificationService/Service/MailgunEmailService.cs [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+                .AddParameter("to", "curiosos-suppport@example.com")
```
Comment (LuisHerbas98) [inline src/DistributionCenter.Services/NotificationService/Service/MailgunEmailService.cs]: and here

**Pull request closed without merging**